packages/app-dashboard/src/components/withdraw/WalletConnect/WalletConnectUtil.ts

```diff
@@ -0,0 +1,290 @@
+import { litNodeClient } from '@/components/consent/utils/lit';
+import { getWalletConnectActions, getCurrentWalletAddress } from './WalletConnectStore';
+
+import { env } from '@/config/env';
+
+const { VITE_WALLETCONNECT_PROJECT_ID } = env;
+
+let pkpWallet: PKPEthersWallet | null = null;
+let walletKitClient: IWalletKit | null = null;
+let isInitializing = false;
+
+/**
+ * Create a PKPEthersWallet using the agent PKP and session signatures
+ * @param agentPKP The agent's PKP to use for signing
+ * @param sessionSigs Session signatures for authentication
+ * @returns The created PKPEthersWallet instance
+ */
+export async function createPKPWallet(
+  agentPKP: IRelayPKP,
+  sessionSigs: SessionSigs,
+): Promise<PKPEthersWallet> {
+  if (!agentPKP?.publicKey) {
+    throw new Error('PKP does not have a public key');
+  }
+
+  pkpWallet = new PKPEthersWallet({
+    pkpPubKey: agentPKP.publicKey,
+    litNodeClient: litNodeClient,
+    controllerSessionSigs: sessionSigs,
+  });
+
+  await pkpWallet.init();
+  return pkpWallet;
+}
+
+/**
+ * Register a PKP wallet with WalletKit to handle signing requests
+ * @param agentPKP The agent's PKP to use for signing
+ * @param sessionSigs Session signatures for authentication
+ * @returns Information about the registered account
+ */
+export async function registerPKPWallet(
+  agentPKP: IRelayPKP,
+  sessionSigs?: SessionSigs,
+): Promise<{ address: string; publicKey: string }> {
+  if (!agentPKP?.ethAddress) {
+    throw new Error('PKP does not have an Ethereum address');
+  }
+
+  const newAddress = agentPKP.ethAddress;
+  const storedWalletAddress = getCurrentWalletAddress();
+  const walletHasChanged = storedWalletAddress !== null && storedWalletAddress !== newAddress;
+
+  const actions = getWalletConnectActions();
+
+  if (walletHasChanged) {
+    console.log(`Wallet has changed from ${storedWalletAddress} to ${newAddress}`);
+
+    // First disconnect any sessions associated with the previous wallet to ensure
+    // dApps know that wallet has been disconnected
+    if (storedWalletAddress) {
+      console.log(`Disconnecting all sessions for previous wallet: ${storedWalletAddress}`);
+      try {
+        await actions.clearSessionsForAddress(storedWalletAddress);
+      } catch (error) {
+        console.error('Error disconnecting sessions for previous wallet:', error);
+      }
+    }
+
+    await resetWalletConnectClient();
+
+    // Update the current wallet address in the store
+    actions.setCurrentWalletAddress(newAddress);
+  } else if (newAddress && storedWalletAddress !== newAddress) {
+    // First time setting this wallet or wallet address isn't tracked yet
+    console.log(`Setting current wallet address to ${newAddress}`);
+    actions.setCurrentWalletAddress(newAddress);
+  }
+
+  // Get or create the client
+  const client = await createWalletConnectClient();
+
+  if (!client) {
+    throw new Error('WalletKit client not initialized');
+  }
```

Comment:
`createWalletConnectclient` returns `Promise<IWalletKit>` so this if does not make sense